📖 Document in place contract (#13297)
kubernetes-sigs/cluster-api: docs/book/src/developer/providers/contracts/bootstrap-config.md

</aside>

### BootstrapConfig: support for in-place changes

In case you are developing an bootstrap config provider with support for in-place updates of the Machine configuration,
you should consider following recommendations during implementation.

- The `Update Extension` is the component responsible for orchestrating in-place changes on Machines.
  Accordingly, the BootstrapConfig controller should ignore in-place changes and do not re-generate the bootstrap config.

> - It might be useful to start thinking at the BootstrapConfig API surface as and set of fields with one of the following behaviors:

Reviewer: ```suggestion
- It might be useful to start thinking about the BootstrapConfig API surface as a set of fields with one of the following behaviors:
```
I think
